src/ErrorHandling/Program.cs

```diff
@@ -0,0 +1,95 @@
+﻿// <copyright file="Program.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+namespace ErrorHandling
+{
+    /// <summary>
+    /// Program class
+    /// </summary>
+    public class Program
+    {
+        /// <summary>
+        /// Main method
+        /// </summary>
+        /// <param name="args">String array of arguments</param>
+        public static void Main(string[] args)
+        {
+            AppDomain.CurrentDomain.UnhandledException += MyHandler;
+            try
+            {
+                int input1 = GetInput("Enter integer 1");
+                int input2 = GetInput("Enter integer 2");
+                Console.WriteLine($"Quotient when {input1} is divided by {input2} is {input1 / input2}");
+                int arraySize = GetInput("Enter the size of the array");
+                CreateAnArray(arraySize);
+            }
+            catch (UserInputInvalidException exception)
+            {
+                Console.WriteLine(exception.Message);
+            }
+            catch (DivideByZeroException exception)
+            {
+                Console.WriteLine(exception.Message);
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine(exception.Message);
+            }
+            finally
+            {
+                Console.WriteLine("Inside finally block");
+            }
+
+            throw new Exception();
+        }
+
+        /// <summary>
+        /// Get input from user and validates it.
+        /// </summary>
+        /// <param name="messageToPrint">Message to print to the user while asking for input</param>
+        /// <returns>Parsed input</returns>
+        private static int GetInput(string? messageToPrint)
+        {
+            Console.WriteLine($"{messageToPrint}");
+            string? input1 = Console.ReadLine();
+            if (!int.TryParse(input1, out int parsedString))
+            {
+                throw new UserInputInvalidException("Invalid input exception, enter only integers.");
```

Comment:
Exception is a development term. Don't use it in user-facing messages. User-facing messages must be clear and crisp.
```suggestion
                throw new UserInputInvalidException("The input value is invalid. Please enter a valid integer.");
```

Reply:
Okay